exotica_core/src/problems/unconstrained_end_pose_problem.cpp

```diff
@@ -93,18 +93,24 @@ Eigen::RowVectorXd UnconstrainedEndPoseProblem::GetScalarJacobian() const
     return cost.jacobian.transpose() * cost.S * cost.ydiff * 2.0;
 }
 
-double UnconstrainedEndPoseProblem::GetScalarTaskCost(const std::string& task_name) const
+Eigen::VectorXd UnconstrainedEndPoseProblem::GetTaskCost(const std::string& task_name) const
```

Comment:
Wouldn't `GetTaskError` or similar not be a more accurate name?

Reply:
I kept costs since it's the internally used name. Do you think it's better to have `GetTaskError` and `GetScalarTaskCost`?

Reply:
You are returning `ydiff` (y*-Phi) which is the TaskError. The cost is ydiff^T * S * ydiff. I would say TaskError here but not change GetScalarCost. Regarding GetTaskCost, I do not think anyone apart from you is using it at the moment? I let @VladimirIvan weigh in as well.

Reply:
Ah yeah, the `EndPoseTask` was called `cost` which contains the `ydiff`. That's why I stayed with `costs`. I am happy to rename if we agree on something.

Reply:
I just saw that `AICOSolver` and `BayesianIKSolver` have a `GetTaskCosts` method. They both return a scalar. Should they maybe be renamed to `GetScalarTaskCost`?

Maybe we should also add `GetScalarTaskCost` as pure virtual to the base class?

Reply:
The EndPoseTask should continue to be called cost. It is after all what it's used for and we should not rename it.

However, your new method wants to return the `ydiff`, i.e., _error_ of the cost. So `GetCostTaskError` could work. The `ScalarCost` is `ydiff^T * S * ydiff` (weighted sum of squares). 

AICO and BayesianIK should be untouched. Their GetTaskCosts are internal/private and used to update the messages. I have a new version of AICO/BayesianIK coming maybe in autumn where these things get revamped.

Reply:
Right, `GetTaskCosts` is private. I overlooked that.

Changed name to `GetTaskError` and added python binding.

Reply:
Thanks - I liked your previous comment so can we go for `GetCostTaskError`?

Reply:
Yes, I force pushed with the new method name `GetTaskError`.